web/src/echo.js

```diff
@@ -400,6 +400,7 @@ export function process_from_server(messages) {
             // the "main" codepath that doesn't have to id reconciliation.
             // We simply return non-echo messages to our caller.
             non_echo_messages.push(message);
+            sent_messages.report_event_received(local_id);
```

Comment:
I think this code path includes messages that were just sent from another browser, in which case I think `local_id` is `undefined`, since we only send `local_id` back to the one client that sent the message.

Reply:
I think this will just silently be a noop if `local_id` is `undefined`, but probably we should do something more clever here.

```
export function report_event_received(local_id) {                                                       
    const state = get_message_state(local_id);                                                          
    if (!state) {                                                                                       
        return;                                                                                         
    }                                                                                                   
                                                                                                        
    state.report_event_received();                                                                      
}                                                                                                       
```

Reply:
This might actually be a more significant bug -- I had a message fail to appear (I think it was not eligible for local echo as it contained a link? Not sure) on chat.zulip.org just now, with this exception in the console:

![image](https://user-images.githubusercontent.com/2746074/236899475-12d726a2-72f6-4e31-b88a-d18c45847fac.png)